js/form_editor_controller.js

```diff
@@ -1106,6 +1114,14 @@ class GlpiFormEditorController
         // Update the question type
         this.#setItemInput(question, "type", type);
 
+        // Check if the form is an anonymous form and if the question type allows it
+        const blacklisted_warning = question.find('[data-glpi-form-editor-blacklisted-question-type-warning]');
+        if (this.#is_anonymous && !this.#options[type].allowAnonymous) {
+            blacklisted_warning.removeClass('d-none');
+        } else {
+            blacklisted_warning.addClass('d-none');
+        }
+
```

Comment:
Do you think this could be done without any javascript ?

I think it should be possible with a specific data tag that is added to anonymous forms and some css rules (like we do for extra data that are hidden/shown by css rules).

Reply:
I tried not to use javascript to manage the element's visibility, but I couldn't find a solution using only CSS/SCSS. Adding a data tag would have necessarily required the use of javascript to make it persist when changing type. However, I removed a lot of javascript code and limited its use as much as possible for this feature.
If you have a better solution, I'd love to hear about it!